### ISSUE TITLE
Falsy Lint/UnusedMethodArgument for specific Ruby 2.3 lonely operator usage

### ISSUE THREAD

**Issue by gsamokovarov** [bbatsov/rubocop#3185]
## Expected behavior

Given the following code below that uses the Ruby 2.3 lonely operator I get a complaint about an unused argument.

```ruby
def self.some_method(bar:)
  something&.some_method(bar) { something_else }
end
```

```
W: Unused method argument - bar
```

Note that it only fails to detect the keyword argument usage if a block is given.

```ruby
def self.some_method(bar:)
  something&.some_method(bar)
end
```

The example above doesn't receive a warning from Rubocop.

## Actual behavior

Don't give unused argument warnings as the arguments are actually used.

## Steps to reproduce the problem

Check out the code snippets above or the patch below.

```patch
diff --git i/spec/rubocop/cop/lint/unused_method_argument_spec.rb w/spec/rubocop/cop/lint/unused_method_argument_spec.rb
index 38f1bf3..48e6203 100644
--- i/spec/rubocop/cop/lint/unused_method_argument_spec.rb
+++ w/spec/rubocop/cop/lint/unused_method_argument_spec.rb
@@ -69,6 +69,18 @@ describe RuboCop::Cop::Lint::UnusedMethodArgument, :config do
       end
     end
 
+    context 'when a required keyword argument is used in lonely operator', ruby: 2.3 do
+      let(:source) { <<-END }
+        def self.some_method(bar:)
+          something&.some_method(bar) { something_else }
+        end
+      END
+
+      it 'accepts' do
+        expect(cop.offenses).to be_empty
+      end
+    end
+
     context 'when an optional keyword argument is unused', ruby: 2 do
       let(:source) { <<-END }
         def self.some_method(foo, bar: 1)
```

## RuboCop version

```
$ rubocop -V
0.40.0 (using Parser 2.3.1.0, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by alexdowad** [comment]
This is a bug in `parser`, not `rubocop`. The AST returned by `parser` is completely wrong.

**Comment by alexdowad** [comment]
Oooh, but it does work with the `parser` gem at `HEAD`!

**Comment by alexdowad** [comment]
Aha, looks like it was fixed in `f08db93`. @whitequark... new minor release please!!!

**Comment by whitequark** [comment]
2.3.1.1 released.

**Comment by alexdowad** [comment]
Thank you, sir!

**Comment by gsamokovarov** [comment]
Thanks, folks!
